sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -176,6 +177,125 @@ case class Like(left: Expression, right: Expression, escapeChar: Char)
   }
 }
 
+abstract class LikeAllBase extends Expression with ImplicitCastInputTypes with NullIntolerant {
+  def value: Expression = children.head
+  def list: Seq[Expression] = children.tail
+  def isNot: Boolean
+
+  override def inputTypes: Seq[AbstractDataType] = {
+    StringType +: Seq.fill(children.size - 1)(StringType)
+  }
+
+  override def dataType: DataType = BooleanType
+
+  override def foldable: Boolean = children.forall(_.foldable)
+
+  override def nullable: Boolean = true
+
+  def matches(regex: Pattern, str: String): Boolean = regex.matcher(str).matches()
+
+  override def eval(input: InternalRow): Any = {
+    val evaluatedValue = value.eval(input)
+    if (evaluatedValue == null) {
+      null
+    } else {
+      var hasNull = false
+      var match = true
+      list.foreach { e =>
+        val str = e.eval(input)
+        if (str == null) {
+          hasNull = true
+        } else {
+          val regex =
+            Pattern.compile(StringUtils.escapeLikeRegex(str.asInstanceOf[UTF8String].toString, '\\'))
+          if ((isNot && matches(regex, evaluatedValue.asInstanceOf[UTF8String].toString)) ||
+            !(isNot || matches(regex, evaluatedValue.asInstanceOf[UTF8String].toString)) {
```

Comment:
can we put `matches` as a local variable to shorten the code here?

Reply:
OK